streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -52,21 +52,27 @@
     private JoinWindows(String name, long before, long after) {
         super(name);
 
+        if (before < 0) {
+            throw new IllegalArgumentException("window size must be > 0 (you provided before as " + before + ")");
+        }
+        if (after < 0) {
+            throw new IllegalArgumentException("window size must be > 0 (you provided after as " + after + ")");
+        }
+        if (before == 0 && after == 0) {
```

Comment:
Any reason why you don't just add these conditions to the if statements above? i.e.,

`if (before <= 0 )`